src/components/typography/heading/heading.tsx

```diff
@@ -1,40 +1,33 @@
 import clsx from 'clsx';
 import { HeadingProps, HeadingSize } from './types';
-import { FC } from 'react';
+import { FC, HTMLAttributes, createElement } from 'react';
 
 export const Heading: FC<HeadingProps> = ({
     children,
-    size,
+    headingLevel,
     className = 'text-inherit',
     ...props
 }) => {
-    const fontClasses = 'font-default leading-s';
-    switch (size) {
-        case HeadingSize.H1:
-            return (
-                <h1 className={clsx(fontClasses, 'text-4xl font-bold', className)} {...props}>
-                    {children}
-                </h1>
-            );
-        case HeadingSize.H2:
-            return (
-                <h2 className={clsx(fontClasses, 'text-3xl font-bold', className)} {...props}>
-                    {children}
-                </h2>
-            );
-        case HeadingSize.H3:
-            return (
-                <h3 className={clsx(fontClasses, 'text-2xl font-semibold', className)} {...props}>
-                    {children}
-                </h3>
-            );
-        case HeadingSize.H4:
-            return (
-                <h4 className={clsx(fontClasses, 'text-xl font-semibold', className)} {...props}>
-                    {children}
-                </h4>
-            );
-        default:
-            return <></>;
+    if (headingLevel && headingLevel === HeadingSize.H1) {
+        className = clsx(className, 'text-4xl font-bold');
+    } else if (headingLevel && headingLevel === HeadingSize.H2) {
+        className = clsx(className, 'text-3xl font-bold');
+    } else if (headingLevel && headingLevel === HeadingSize.H3) {
+        className = clsx(className, 'text-2xl font-bold');
+    } else if (headingLevel && headingLevel === HeadingSize.H4) {
+        className = clsx(className, 'text-xl font-semibold');
+    } else {
+        className = clsx(className, '');
```

Comment:
use clsx() here to determine the classes, example:
`clsx(
            `hover:ring-3 active:ring-4`,
            {
                primary: `bg-primary-600 fill-white text-white hover:bg-primary-700 hover:ring-primary-100 active:bg-primary-700 active:ring-primary-200`,
                secondary: `bg-secondary-600 fill-white text-white hover:bg-secondary-700 hover:ring-secondary-100 active:bg-secondary-700 active:ring-secondary-200`,
                tertiary:
                    'bg-gradient-to-r from-tertiary-500 to-primary-500 fill-white text-white hover:bg-gradient-30-70 hover:ring-primary-100 active:bg-gradient-20-80 active:ring-primary-200',
                quaternary:
                    'bg-secondary-300 fill-secondary-500 text-secondary-500 hover:bg-secondary-200 hover:ring-secondary-100 active:bg-secondary-200 active:ring-secondary-200',
            }[variant],`